### PR TITLE
Only read file contents after checking extension

### LINKED ISSUE #55: [Bug report] Large files (not markdown) cause the plugin to crash
**Description**

I came across an issue wherein the plugin was trying to read the contents of a large .7z file (which is obviously not in the extension whitelist), hitting a limit in V8 itself, and stopping the plugin.

This is fixed by #54, simply by only reading the contents _after_ checking the extension.

### PR DESCRIPTION
**Pull request by RLGingerBiscuit**
Fixes #55.

### CODE REVIEW

**Comment by isaaclyman** [comment]
Thanks for the PR! This is a good fix. I have just a couple questions, to follow.

File: logic/file.ts [hunk 1]
```diff
@@ -220,10 +221,11 @@ export class FileHelper {
 		};
 
 		const metadata = this.app.metadataCache.getFileCache(file) as CachedMetadata | null;
```
Comment (isaaclyman) [inline logic/file.ts:223]: Is there a reason we can't look at the metadataCache for a file that shouldn't be counted?

"Shouldn't be counted" in the context of this plugin means no character count, word count, or page count. Users may still like to see Created Date, Last Updated Date, or File Size, all of which live in the metadata.

**Comment by isaaclyman** [comment]
I'd like to test my changes in commit d7d4087 on your big 7z file. Do you mind fetching the commit and taking a look? Or providing a comparable file I can include in my test vault?

File: logic/file.ts (hunk 1 above)
Comment (isaaclyman) [reply logic/file.ts:223]: Correction: those live in `file.stat`, not the metadata. You were right here, please disregard.

Comment (isaaclyman) [reply logic/file.ts:223]: Okay, so aside from a couple of CPU cycles checking that there's no metadata for the file, it seems like there wouldn't be any harm in leaving the shouldCountFile check as-is. But either way, this code does prevent tracking dates and file size for files with disallowed extensions. I'll publish my latest changes (based largely on this PR, but with a couple of small differences) and await your feedback. Thanks!

**Comment by RLGingerBiscuit** [comment]
Yup, that appears to fix the issue. My vault now works correctly with this plugin and it's set accordingly in the cache. Thanks.

**Comment by isaaclyman** [comment]
Awesome. Thanks again for your help.

**Pull request closed without merging**